airlock_processor/BlobCreatedTrigger/__init__.py

```diff
@@ -32,10 +32,11 @@ def main(msg: func.ServiceBusMessage,
             logging.error("environment variable 'ENABLE_MALWARE_SCANNING' does not exists. Cannot continue.")
             raise
 
-        if enable_malware_scanning:
+        if enable_malware_scanning and constants.STORAGE_ACCOUNT_NAME_IMPORT_INPROGRESS in topic:
             # If malware scanning is enabled, the fact that the blob was created can be dismissed.
             # It will be consumed by the malware scanning service
             logging.info('Malware scanning is enabled. no action to perform.')
+            send_delete_event(dataDeletionEvent, json_body, request_id)
```

Comment:
I have tested this version and it's triggering a runtime error. The blob storage inside `stalimip<TRE_ID>` is being deleted twice. I could stop getting such errors moving this line to a `if` statement. Something like this:

```
if event_type is None or event_type != 'Microsoft.Storage.BlobDeleted':
    send_delete_event(dataDeletionEvent, json_body, request_id)
return
```

A few lines above I checked if `eventType` is in the `json_body`, and if so, I assigned it the value store in the `json_body`.

I believe the deletion occur because, 1) the file is moved from storage account `stalimip<TRE_ID>` to a storage account inside the workspace, and it's the last blob in the storage account, and 2) the request status changed (I couldn't identify the correct order of execution).

Reply:
@migldasilva  I will test this in a separate PR. Will try to reproduce on my end, since this shouldnt be triggered from a 'Blob.Deleted' event....